client/asset/dcr/dcr.go

```diff
@@ -3538,6 +3541,9 @@ func (dcr *ExchangeWallet) lookupTxOutput(ctx context.Context, txHash *chainhash
 	if confs == 0 {
 		// Only counts as spent if spent in a mined transaction,
 		// unconfirmed tx outputs can't be spent in a mined tx.
+		if dcr.wallet.SpvMode() {
+			return txOut, confs, -1, nil // make it do a cfilters scan because GetTransaction lies
+		}
```

Comment:
This is a possible workaround, which unfortunately forces the cfilters scan path even when `GetTransaction` did return a result (it's a wallet transaction i.e. checking your own swap confirms or it's a self-trade)